changelogs/unreleased/5508-move-increment-calculation-to-release-version-pipeline.yml

```diff
@@ -0,0 +1,7 @@
+description: Move increment calculation to release_version pipeline to prevent resource state change outside of the agent's deploy lock.
```

Comment:
I don't think this changelog message is very useful for an end-user. It only makes sense if you know about how the orchestrator works internally. The end-user is more interested in knowing which issue exists, when it can occur and how big the impact is. From that perspective it would be better to write something like this: `Fix race condition that causes the deployment of a resource to fail with the error message: "Fetching resource events only makes sense when the resource is currently deploying". This issue happens in rare situation and the orchestrator will automatically recover from it during the next deployment.`

However, I do like the fact that the changelog contains a short description regarding the technical implementation that was done to resolve the problem. The good news is that our changelog files supports this. We could do the following:

```
description: Move increment calculation to release_version pipeline to prevent resource state change outside of the agent's deploy lock.
change-type: minor
destination-branches:
  - master
  - iso6
sections:
  bugfix: "Fix race condition that causes the deployment of a resource to fail with the error message: "Fetching resource events only makes sense when the resource is currently deploying". This issue happens in rare situation and the orchestrator will automatically recover from it during the next deployment."
```
The `bugfix` section will go into the changelog and the description field is there for internal use.